src/cryptography/x509/extensions.py

```diff
@@ -1446,6 +1446,145 @@ def __repr__(self):
     nonce = utils.read_only_property("_nonce")
 
 
+@utils.register_interface(ExtensionType)
+class IssuingDistributionPoint(object):
+    oid = ExtensionOID.ISSUING_DISTRIBUTION_POINT
+
+    def __init__(self, only_contains_user_certs, only_contains_ca_certs,
+                 indirect_crl, only_contains_attribute_certs,
+                 only_some_reasons, full_name, relative_name):
+
+        if (
+            only_some_reasons and (
+                not isinstance(only_some_reasons, frozenset) or not all(
+                    isinstance(x, ReasonFlags) for x in only_some_reasons
+                )
+            )
+        ):
+            raise TypeError(
+                "only_some_reasons must be None or frozenset of ReasonFlags"
+            )
+
+        if only_some_reasons and (
+            ReasonFlags.unspecified in only_some_reasons or
+            ReasonFlags.remove_from_crl in only_some_reasons
+        ):
+            raise ValueError(
+                "unspecified and remove_from_crl are not valid reasons in an "
+                "IssuingDistributionPoint"
+            )
+
+        if not (
+            isinstance(only_contains_user_certs, bool) and
+            isinstance(only_contains_ca_certs, bool) and
+            isinstance(indirect_crl, bool) and
+            isinstance(only_contains_attribute_certs, bool)
+        ):
+            raise TypeError(
+                "only_contains_user_certs, only_contains_ca_certs, "
+                "indirect_crl and only_contains_attribute_certs "
+                "must all be boolean."
+            )
+
+        bools = [
+            only_contains_user_certs, only_contains_ca_certs,
+            indirect_crl, only_contains_attribute_certs
+        ]
+
+        if len([x for x in bools if x is True]) > 1:
```

Comment:
I think this is actually more clearly written as `if only_contains_user_certs + only_contains_ca_certs + indirect_crl + only_contains_attribute_certs > 1`?

Reply:
Isn't that abusing the fact that a bool is secretly an int? `True + True + False == 2` is kind of a Python WTF

Reply:
Yes, it is. If you don't like that, at least drop the `is True` and just make this `if x`. And rename `bools` to something semantically meaningful.